File: backend/src/test/java/com/baltic/road/path/app/extensions/Book.java
package com.baltic.road.path.app.extensions;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.PARAMETER)
@Retention(RetentionPolicy.RUNTIME)
public @interface Book {
    String id();
    String token();
    String title();
}

File: backend/src/main/kotlin/com/baltic/road/path/app/gr/model/api/Book.kt
package com.baltic.road.path.app.gr.model.api

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty

data class Book(
    @JacksonXmlProperty(isAttribute = true)
    var type: String,
    var id: Id,
    var title: String,
    @JacksonXmlElementWrapper(useWrapping = false)
    var author: List<Author> = ArrayList()
)
